### PR TITLE
add column for checkboxes/radiobuttons/menu icons

### PR DESCRIPTION
**Pull request by fkloft**
Add a column to show the current state for checkboxes and radiobuttons.

I also wanted to include menu icons, but since `Command` is exported via DBus, I didn't know how to declare functions with unserializable parameters/return values. I have no experience with Vala, but I did my best to write proper code, including null checks.

screenshot

### CODE REVIEW

**Comment by fkloft** [comment]
I managed to get the icons to show up as well.

screenshot from nemo

**Comment by p-e-w** [comment]
These features look really cool at first, so it is with a heavy heart that I must tell you that I will not merge this PR. I had considered adding both check/radio indicators and icons myself in the past, and ultimately decided against it.

First, the radio buttons. Those just don't make sense when combined with a filterable list. The purpose of radio buttons is to represent a group of mutually exclusive options. Filtering and re-sorting destroys that group association, and can give the confusing impression that items from multiple radio groups are mutually exclusive, or that multiple radio buttons from the same group are active at the same time.

Menu item icons are something I associate with the early 2000s, and I believe it's for a good reason that GNOME hides them by default now. Menus are a list of text options. Action icons are for buttons. If this UI split had been clearer from the start, something like Plotinus might have been built into GTK+ years ago, since text = searchable.

This leaves only checkboxes then. I do see some value in indicating state for those, but I really don't like the idea of adding another column *just* for checkboxes. Also, indicating check but not radio state makes no sense. So overall, I prefer the layout to stay as it is now.

I really am very sorry to reject this effort so outright. It is for reasons such as these that the README request any additions be discussed in an issue before creating a PR.

**Comment by fkloft** [comment]
I can see your point, but I tend to disagree.
- regarding radio buttons: menus can have radio buttons on them right next
to other options. Nemo for example has (symbols|list|compact) at the end of
the "View" menu, below the other options. I think that the very point of
plotinus is to show an entire menu hierarchy in a flattened list, so it
should be obvious that it may contain multiple radio groups. Especially
during filtering, I really find it useful to see which option is currently
active.
- regarding icons: I know the gnome ideology on menu icons, and I've
implemented their guidelines: icons should only represent nouns (like
places and bookmarks in Nemo), but not verbs (like copy, new, delete). My
version only displays icons when the menu has forced them to display (which
according to gnome should only be used for nouns) or if the user whishes it
(and has changed their gtk settings).

Would you consider this if it was an optional feature?

**Comment by TLogiviere** [comment]
In case my opinion can help on this issue, listing check-boxes can be seen as a free improvement to the usefulness of this cool piece of software.
1. They aren't subject to semantic destruction when searched through,
2. Listing them helps plotinus goal of allowing fast gui manipulation which is kinda its reason of existing.

Then about radios, p-e-w made a point and my p.o.v. is that this widget shouldn't be listed at all OR (because, as fkloft stated it can be usefull) by considering semantic groups as one immutable entity and display it in totality whenever at least one option is matched by a search.

Finally, about icons and after trying your version, fkloft, there is at least one problem which probably is a corner case : some window states are expressed by check-boxes e.g. « full screen » option in gedit.
In that case Plotinus ends-up showing an icon to the side of the widget and as a result, the interface looks bloated by unnecessary information as the icon, not following the state of the window (contrary to the gui button), does not gives a better explanation than the text or more information than the widget.
In such cases it would probably be better to only display the widget as it carries the information the user is interested in.
Interestingly in nautilus command pallet, there is no icon displayed for the bookmark action, only the text, even if the interface of nautilus show it only with an icon. Rightfully though, as there does not seems to be matter for ambiguity given the fact that the button disappear/is greyed out when a bookmark already exists.

tl;dr: check-boxes add useful, free of charge information ; group of radio buttons should be considered as an unsplitable entity ; state widgets should be prioritized over icons.

**Comment by tallero** [comment]
I think it is fun to note that gnome mockups for palette include icons for actions:
https://gitlab.gnome.org/Teams/Design/os-mockups/raw/master/command-search/command-search.png
